internal/attestation/measurements/measurement-generator/generate.go

```diff
@@ -0,0 +1,256 @@
+	warnOnlyColon := token.Pos(warnOnlyKeyPos + 9)
+	warnOnlyValuePos := token.Pos(warnOnlyColon + 2)
+	var rbrace token.Pos
+	if measuremnt.WarnOnly {
+		rbrace = token.Pos(warnOnlyValuePos + 9) // 9 = len("true") + padding
+	} else {
+		rbrace = token.Pos(warnOnlyValuePos + 10) // 10 = len("false") + padding
+	}
+
+	return &ast.KeyValueExpr{
+		Key: &ast.BasicLit{
+			ValuePos: pos,
+			Kind:     token.INT,
+			Value:    key,
+		},
+		Colon: colon,
+		Value: &ast.CompositeLit{
+			// Type:   ast.NewIdent("Measurement"),
+			Lbrace: valuePos,
+			Elts: []ast.Expr{
+				&ast.KeyValueExpr{
+					Key:   &ast.Ident{NamePos: expectedKeyPos, Name: "Expected"},
+					Colon: expectedColon,
+					Value: byteArrayCompositeLit(measuremnt.Expected, expectedValuePos),
+				},
+				&ast.KeyValueExpr{
+					Key:   &ast.Ident{NamePos: warnOnlyKeyPos, Name: "WarnOnly"},
+					Colon: warnOnlyColon,
+					Value: &ast.Ident{NamePos: warnOnlyValuePos, Name: strconv.FormatBool(measuremnt.WarnOnly)},
+				},
+			},
+			Rbrace: rbrace,
+		},
+	}, rbrace + 1
+}
+
+func measurementsCompositeLiteral(measurements measurements.M, pos token.Pos) *ast.CompositeLit {
+	lbrace := pos + 1
+	var elts []ast.Expr
+	pcrs := make([]uint32, 0, len(measurements))
+	for pcr := range measurements {
+		pcrs = append(pcrs, pcr)
+	}
+	sort.Slice(pcrs, func(i, j int) bool { return pcrs[i] < pcrs[j] })
+	entryPos := lbrace + 5
+	for _, pcr := range pcrs {
+		kvExpr, newEntryPos := measurementsEntryKeyValueExpr(pcr, measurements[pcr], entryPos)
+		elts = append(elts, kvExpr)
+		entryPos = newEntryPos + 4 // 4 = padding and newline
+	}
+	return &ast.CompositeLit{
+		Lbrace: lbrace,
+		Type: &ast.Ident{
+			NamePos: pos,
+			Name:    "M",
+		},
+		Elts:   elts,
+		Rbrace: entryPos - 1,
+	}
+}
+
+func main() {
+	defaultConf := config.Default()
+	log.Printf("Generating measurements for %s\n", defaultConf.Image)
+
+	const filePath = "./measurements_enterprise.go"
+
+	ctx := context.Background()
+	fset := token.NewFileSet()
+	file, err := parser.ParseFile(fset, filePath, nil, parser.ParseComments)
+	if err != nil {
+		log.Fatal(err)
+	}
+	rekor, err := sigstore.NewRekor()
+	if err != nil {
+		log.Fatal(err)
+	}
+	newFile := astutil.Apply(file, func(cursor *astutil.Cursor) bool {
+		n := cursor.Node()
+
+		if clause, ok := n.(*ast.CaseClause); ok && len(clause.List) > 0 && len(clause.Body) > 0 {
+			sel, ok := clause.List[0].(*ast.SelectorExpr)
+			if !ok {
+				return true
+			}
+			returnStmt, ok := clause.Body[0].(*ast.ReturnStmt)
+			if !ok || len(returnStmt.Results) == 0 {
+				return true
+			}
+
+			provider := cloudprovider.FromString(sel.Sel.Name)
+			if provider == cloudprovider.Unknown {
+				return true
+			}
+			log.Println("Found", provider)
+			measuremnts := mustGetMeasurements(ctx, rekor, []byte(constants.CosignPublicKey), http.DefaultClient, provider, defaultConf.Image)
+			returnStmt.Results[0] = measurementsCompositeLiteral(measuremnts, returnStmt.Return+7)
```

Comment:
Please add some type of feedback like a counter so we know if we found any measurements at all. Have a look at the updated hash-generator.